src/idl/src/annotation.c

```diff
@@ -187,47 +183,61 @@ annotate_extensibility(
   idl_annotation_appl_t *annotation_appl,
   idl_node_t *node)
 {
-  idl_extensibility_t extensibility = IDL_EXTENSIBILITY_FINAL;
+  const idl_annotation_appl_t **annotationp = NULL;
+  idl_extensibility_t *extensibilityp = NULL, extensibility = IDL_FINAL;
   const char *annotation;
 
   assert(annotation_appl);
   annotation = idl_identifier(annotation_appl->annotation);
   assert(annotation);
 
   if (strcmp(annotation, "final") == 0) {
-    extensibility = IDL_EXTENSIBILITY_FINAL;
+    extensibility = IDL_FINAL;
   } else if (strcmp(annotation, "appendable") == 0) {
-    extensibility = IDL_EXTENSIBILITY_APPENDABLE;
+    extensibility = IDL_APPENDABLE;
   } else if (strcmp(annotation, "mutable") == 0) {
-    extensibility = IDL_EXTENSIBILITY_MUTABLE;
+    extensibility = IDL_MUTABLE;
   } else if (strcmp(annotation, "extensibility") == 0) {
-    const char *enumerator = "";
+    const char *enumerator;
 
     assert(annotation_appl->parameters);
     assert(annotation_appl->parameters->const_expr);
     enumerator = idl_identifier(annotation_appl->parameters->const_expr);
     assert(enumerator);
     if (strcmp(enumerator, "FINAL") == 0) {
```

Comment:
case (in)sensitivity isn't really relevant here. The annotation applications refer to built-in annotations and so always use the same case. The process of looking the annotation identifiers up is case (in)sensitive based on the respective parser option.

Reply:
My "case insensitive" came about because of the different cases used in the string literals a few lines apart. I suppose what you say is that it would have to be a flag indicating whether the string is expected to be in lower case or in upper case, but that doesn't change the observation that reducing the code duplication here *may* be worth the effort.

Reply:
I see what you mean. Definitely, checking for extensibility first and making the rest case-insensitive seems like a better approach.

Reply:
Done.